nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,611 @@
+            following, unit_path, job_id, job_type, job_path) = unit
+        if following != '':
+            continue
+        if job_id == 0 and active_state == 'inactive':
+            continue
+        res[UnitName(id)] = ActiveUnit(load=load_state,
+                                       state=active_state,
+                                       sub_state=sub_state)
+
+    return res
+
+Mountpoint = NewType('Mountpoint', str)
+BlockDevice = NewType('BlockDevice', str)
+FsMount = NamedTuple('FsMount', [('device', BlockDevice),
+                                 ('fs_type', str),
+                                 ('options', str)])
+SwapMount = NamedTuple('SwapMount', [('options', str)])
+
+def parse_fstab(filename: Path) -> Tuple[Dict[Mountpoint, FsMount], Dict[BlockDevice, SwapMount]]:
+    fss = {}  # type: Dict[Mountpoint, FsMount]
+    swaps = {}  # type: Dict[BlockDevice, SwapMount]
+    for line in filename.read_text().split('\n'):
+        line = line.strip()
+        line = re.sub(r'^\s*#.*', '', line)
+        if line.strip() == '':
+            continue
+
+        xs = re.split(' ', line)
+        device = BlockDevice(xs[0])
+        mountpoint = Mountpoint(xs[1])
+        fs_type = xs[2]
+        options = xs[3] if len(xs) > 4 else ''
+        if fs_type == 'swap':
+            swaps[device] = SwapMount(options)
+        else:
+            fss[mountpoint] = FsMount(device, fs_type, options)
+
+    return (fss, swaps)
+
+def parse_unit(filename: Path) -> Dict[str, str]:
+    info = {}
+    info.update(parse_key_values(read_if_exists(filename)))
+    overrides = filename.with_name(filename.name + ".d") / "overrides.conf"
+    info.update(parse_key_values(read_if_exists(overrides)))
+    return info
+
+def parse_key_values(s: str) -> Dict[str, str]:
+    info = {}
+    for line in s.split('\n'):
+        m = re.match(r'^([^=]+)=(.*)$', line)
+        if m is None:
+            die('error parsing unit')
+        else:
+            info[m.group(1)] = m.group(2)
+
+    return info
+
+def bool_is_true(s: str) -> bool:
+    return s in ['yes', 'true']
+
+# As a fingerprint for determining whether a unit has changed, we use
+# its absolute path. If it has an override file, we append *its*
+# absolute path as well.
+def fingerprint_unit(filename) -> str:
+    fprint = str(filename.abspath())
+    override = filename.with_name(filename.name+".d") / "overrides.conf"
+    if override.is_file():
+        fprint += ' ' + override.abspath()
+    return fprint
+
+class UnitTransitions:
+    def __init__(self):
+        def read_unit_list(fname: Path) -> Set[UnitName]:
+            units = read_if_exists(fname, "").split('\n')
+            return { UnitName(unit) for unit in units }
+
+        self.to_stop = set()
+        self.to_skip = set()
+        self.to_start = read_unit_list(start_list_file)
+        self.to_restart = read_unit_list(restart_list_file)
+        self.to_reload = read_unit_list(reload_list_file)
```

Comment:
Shouldn't it be START_UNIT_FILE (line 16) and so on ?

Reply:
Quite right. Looks like I missed a type annotation as well.